modules/vm/templates/entrypoint.sh.tpl

```diff
@@ -11,8 +11,82 @@ until ping -c 1 google.com &> /dev/null; do
+# Find/create/attach volumes
+
+instanceHostname=\'$(hostname)\'
+subscriptionID=$(az account show --query "id" --output tsv)
+resourceGroup=$(az vmss list --query "[0].resourceGroup" --output tsv)
+vmssName=$(az vmss list --query "[0].name" --output tsv)
+instanceID=$(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?contains(osProfile.computerName, ${instanceHostname})].instanceId" --output tsv)
+zoneID=$(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?contains(osProfile.computerName, ${instanceHostname})].zones" --output tsv)
+regionID=$(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?contains(osProfile.computerName, ${instanceHostname})].location" --output tsv)
+# TODO replace with external variables
+tier="P40"
+lun=2 # device will be mounted as sdb
+graphdb_device="/dev/sdb"
+diskSizeGB=128
+
+# Define the disk name based on the hostname
+diskName="Disk_${vmssName}_${instanceID}"
+
+# Check if the device sdb exists
+if [ ! -b /dev/sdb ]; then
+    echo "Device /dev/sdb not found."
+
+    # Check if the managed disk is already attached to this instance
+    if [[ $(az vmss list-instances --resource-group $resourceGroup --name $vmssName --query "[?instanceId==\"$instanceID\"].disks[].name" --output tsv) == "$diskName" ]]; then
+        echo "Managed disk is already attached."
+    else
+        # Try to attach an existing managed disk
+        availableDisk=$(az disk list --resource-group $resourceGroup --query "[?diskState=='Unattached' && starts_with(name, 'Disk_${vmssName}') && zones[0]=='${zoneID}'].{Name:name}" --output tsv)
+        if [ -n "$availableDisk" ]; then
+            echo "Attaching available disk $availableDisk."
+            az vmss disk attach --vmss-name $vmssName --resource-group $resourceGroup --instance-id $instanceID --lun $lun --disk "$availableDisk"
+        else
+            # Create a new managed disk
+            echo "Creating a new managed disk and attaching it."
+            az disk create --resource-group $resourceGroup --name $diskName --size-gb $diskSizeGB --location $regionID --sku Premium_LRS --zone $zoneID --tier $tier
+
+            # Attach the newly created disk
+            az vmss disk attach --vmss-name $vmssName --resource-group $resourceGroup --instance-id $instanceID --lun $lun --sku Premium_LRS --disk "/subscriptions/$subscriptionID/resourceGroups/$resourceGroup/providers/Microsoft.Compute/disks/$diskName"
+        fi
+    fi
+fi
+
+# Check if the device /dev/sdb exists again after attaching or creating the disk
+if [ -b /dev/sdb ]; then
+    echo "Device /dev/sdb is available."
+else
+    echo "Device /dev/sdb is not available. Something went wrong."
+fi
+
+# create a file system if there isn't any
+if [ "$graphdb_device: data" = "$(file -s $graphdb_device)" ]; then
+  mkfs -t ext4 $graphdb_device
+fi
+
+disk_mount_point="/var/opt/graphdb"
```

Comment:
Have to return the permissions back to the graphdb user, after all the attaching

Reply:
Actually after the mounting. I should also create the node and proxy folders. 
Added